crates/hyperqueue/src/server/client/mod.rs

```diff
@@ -353,12 +353,22 @@ fn reconstruct_historical_events(
                     ));
                 }
                 JobTaskState::Canceled { cancelled_date, .. } => {
-                    events.push(Event::at(
-                        *cancelled_date,
-                        EventPayload::TaskCanceled {
-                            task_id: TaskId::new(job.job_id, *id),
-                        },
-                    ));
+                    if let Some(task_ids) = events.last_mut().and_then(|e| {
```

Comment:
Why is this needed if the server should now generate cancels in batches?

Reply:
I wasn't sure why we needed to do the batching here, but now I understand, thanks.